lib/publish.js

```diff
@@ -77,10 +77,12 @@ module.exports = async (pluginConfig, context) => {
       description: notes && notes.trim() ? notes : gitTag,
       milestones,
       assets: {
-        links: assetsList.map(({label, alt, url}) => {
+        links: assetsList.map(({label, alt, url, type, filepath}) => {
           return {
             name: label || alt,
             url: urlJoin(gitlabUrl, repoId, url),
+            link_type: type,
+            filepath,
```

Comment:
One small issue I discovered while testing this branch in a test project. If `.releaserc.json` defines its assets using a wildcard in the `path`:

```json
"assets": [
  {
    "path": "images/**/*",
    "type": "image",
    "filepath": "/path/to/image"
  }
]
```

and if the wildcard matches more than one file, GitLab will throw an error when creating the release. This is because behind the scenes, this plugin creates a separate asset link entry for each matched file, and each entry will have the same `type` and `filepath`. However, Gitlab requires that the `filepath` be unique in the context of a release.

Unfortunately this causes kind of a generic error message when running `semantic-release` - it wouldn't be very clear to users what's going wrong.

So, I think we should:

1. Document this limitation
2. Validate against this and throw a helpful error message so that it's obvious to end users what's going wrong and how to fix it

I think the easiest way to validate this would be to add a validator to `verify.js` that disallows the `filepath` keyword for asset definitions with a `*` in the `path`.

Another option might be to throw an error in `publish.js` after we've expanded the `path` pattern. This would allow the `filepath` option to succeed provided the `path` only resolves to a single file. But I think this might be a source for frustration - in the future, the path may start to match more than one file, which would cause the pipeline to break. So I think the `verify.js` option is our best bet.

Are you up for taking this on? Let me know if you need help.

Reply:
I can spare time to fix this in 2 days or so. I will also include some more tests.